@xen-orchestra/lite/src/components/TreeItem.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import styled from 'styled-components'
+import { withState } from 'reaclette'
+import { IconName } from '@fortawesome/fontawesome-svg-core'
+
+import Icon from './Icon'
+
+interface ParentState {}
+
+interface State {
+  selectedNodeId: string | undefined
+}
+
+interface Props {
+  icon?: IconName
+  id: string
+  // isOpened: boolean
+  label: string
+  children?: React.ReactNode
+}
+
+interface ParentEffects {}
+
+interface Effects {
+  setSelectedNodeId: (id: string) => void
+}
+
+interface Computed {}
+
+const StyledUl = styled.ul`
+  list-style-type: none;
+  cursor: grab;
+`
+
+const TreeItem = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      selectedNodeId: undefined,
+    }),
+    effects: {
+      setSelectedNodeId: function (id) {
+        if (this.state.selectedNodeId === undefined) {
+          this.state.selectedNodeId = id
+          return
+        }
+        this.state.selectedNodeId = undefined
+      },
```

Comment:
Why don't use a function that `toggle` a boolean stored in the state ?

Reply:
done